easybuild/tools/filetools.py

```diff
@@ -1381,14 +1381,21 @@ def apply_regex_substitutions(paths, regex_subs, backup='.orig.eb'):
                     txt_utf8 = txt.decode(encoding='utf-8', errors='replace')
                     write_file(path, txt_utf8)
 
-                for line_id, line in enumerate(fileinput.input(path, inplace=1, backup=backup_ext)):
-                    for regex, subtxt in compiled_regex_subs:
-                        match = regex.search(line)
-                        if match:
-                            origtxt = match.group(0)
-                            _log.info("Replacing line %d in %s: '%s' -> '%s'", (line_id + 1), path, origtxt, subtxt)
-                        line = regex.sub(subtxt, line)
-                    sys.stdout.write(line)
+                backup_path = path + (backup or '.bak')
```

Comment:
The `or '.bak` part is new, and implies always backing up?

That's a change in behavior, see code removed above that mentions *no* backup is created when an empty string is passed in as `backup` value...

I understand this complicates things below w.r.t. "streaming", but we should be careful here imho, in some places (like easyblocks beyond our control), the may rely on the `backup=''` behavior...

We can grab the text when doing the `.read()` above?

Reply:
Yes and no. I basically copied the logic from fileinput which does the same. The backup is removed if it is not wanted later but we need it for the stream-processing. So there is no change in behavior in practice: It is just more visible what also happened before

Yes using what we read above makes sense. This would likely be even faster. Maybe. However this means we'd need to do the line splitting ourselves. Not sure how different line endings are handled by the file iterable. So maybe move that to an issue and do later? As mentioned: No behavior change by now here.

Reply:
Ok it seems we can split the lines simply be `\n` and not reread the file. Sounds easy, will try.